api/src/main/java/org/apache/commons/rdf/api/RDFSyntax.java

```diff
@@ -22,6 +22,124 @@
 import java.util.Optional;
 
 /**
+ * An RDF syntax, e.g. as used for parsing and writing RDF.
+ * <p>
+ * An RDF syntax is uniquely identified by its {@link #mediaType()}, and has a
+ * suggested {@link #fileExtension()}.
+ * <p>
+ * Some of the RDF syntaxes may {@link #supportsDataset()}, meaning they can
+ * represent {@link Quad}s.
+ * <p>
+ * An enumeration of the official RDF 1.1 syntaxes is available in 
+ * {@link OfficialRDFSyntax} - for convenience they are also accessible
+ * as constants here, e.g. <code>RDFSyntax.JSONLD</code>.
+ * 
+ */
+public interface RDFSyntax {
+ 
+    public static OfficialRDFSyntax JSONLD = OfficialRDFSyntax.JSONLD;
+    public static OfficialRDFSyntax TURTLE = OfficialRDFSyntax.TURTLE;
+    public static OfficialRDFSyntax NQUADS = OfficialRDFSyntax.NQUADS;
+    public static OfficialRDFSyntax NTRIPLES = OfficialRDFSyntax.NTRIPLES;
+    public static OfficialRDFSyntax RDFA_HTML = OfficialRDFSyntax.RDFA_HTML;
+    public static OfficialRDFSyntax RDFA_XHTML = OfficialRDFSyntax.RDFA_XHTML;
+    public static OfficialRDFSyntax RDFXML = OfficialRDFSyntax.RDFXML;
+    public static OfficialRDFSyntax TRIG = OfficialRDFSyntax.TRIG;
+    
+    /**
+     * A short name of the RDF Syntax.
+     * <p>
+     * The name typically corresponds to the {@link Enum#name()} of for
+     * {@link OfficialRDFSyntax}, e.g. <code>JSONLD</code>.
+     * 
+     * @return Short name for RDF syntax
+     */
+    public String name();
+
+    /**
+     * The title of the RDF Syntax.
+     * <p>
+     * This is generally the title of the corresponding standard, 
+     * e.g. <em>RDF 1.1 Turtle</em>.
+     * 
+     * @return Title of RDF Syntax
+     */
+    public String title();    
+    
+    /**
+     * The <a href="https://tools.ietf.org/html/rfc2046">IANA media type</a> for
+     * the RDF syntax.
+     * <p>
+     * The media type can be used as part of <code>Content-Type</code> and
+     * <code>Accept</code> for <em>content negotiation</em> in the
+     * <a href="https://tools.ietf.org/html/rfc7231#section-3.1.1.1">HTTP
+     * protocol</a>.
+     */
+    public String mediaType();
+
+    /**
+     * The <a href="https://tools.ietf.org/html/rfc2046">IANA-registered</a>
+     * file extension.
+     * <p>
+     * The file extension includes the leading period, e.g. <code>.jsonld</code>
+     */
+    public String fileExtension();
+
+    /**
+     * Indicate if this RDF syntax supports
+     * <a href="https://www.w3.org/TR/rdf11-concepts/#section-dataset">RDF
+     * Datasets</a>.
+     */
+    public boolean supportsDataset();
+
+
```

Comment:
I think that is out of scope for now, as Commons RDF does not specify namespace preservation yet anywhere else.

Let's revisit that when we make `RDFWriter` where it might be a bit more natural than in `RDFParser` (although I know RDF4J can preserve parsed namespaces)